ml-algorithms/src/main/java/org/opensearch/ml/engine/agents/AgentContextUtil.java

```diff
@@ -0,0 +1,201 @@
+
+        String systemPrompt = parameters.get(SYSTEM_PROMPT_FIELD);
+        if (systemPrompt != null) {
+            builder.systemPrompt(systemPrompt);
+        }
+
+        String userPrompt = parameters.get(QUESTION);
+        if (userPrompt != null) {
+            builder.userPrompt(userPrompt);
+        }
+
+        if (toolSpecs != null) {
+            builder.toolConfigs(toolSpecs);
+        }
+
+        Map<String, String> contextParameters = new HashMap<>();
+        contextParameters.putAll(parameters);
+        contextParameters.put("_current_tool_output", toolOutput);
+        builder.parameters(contextParameters);
+
+        return builder.build();
+    }
+
+    public static Object extractProcessedToolOutput(ContextManagerContext context) {
+        if (context.getParameters() != null) {
+            return context.getParameters().get("_current_tool_output");
+        }
+        return null;
+    }
+
+    public static Object extractFromContext(ContextManagerContext context, String key) {
+        if (context.getParameters() != null) {
+            return context.getParameters().get(key);
+        }
+        return null;
+    }
+
+    public static ContextManagerContext buildContextManagerContext(
+        Map<String, String> parameters,
+        List<String> interactions,
+        List<MLToolSpec> toolSpecs,
+        Memory memory
+    ) {
+        ContextManagerContext.ContextManagerContextBuilder builder = ContextManagerContext.builder();
+
+        String systemPrompt = parameters.get(SYSTEM_PROMPT_FIELD);
+        if (systemPrompt != null) {
+            builder.systemPrompt(systemPrompt);
+        }
+
+        String userPrompt = parameters.get(QUESTION);
+        if (userPrompt != null) {
+            builder.userPrompt(userPrompt);
+        }
+
+        if (memory instanceof ConversationIndexMemory) {
+            String chatHistory = parameters.get(CHAT_HISTORY);
+            if (chatHistory != null) {
+                List<Interaction> chatHistoryList = new ArrayList<>();
+                builder.chatHistory(chatHistoryList);
+            }
+        }
+
+        if (toolSpecs != null) {
+            builder.toolConfigs(toolSpecs);
+        }
+
+        List<Map<String, Object>> toolInteractions = new ArrayList<>();
+        if (interactions != null) {
+            for (String interaction : interactions) {
+                Map<String, Object> toolInteraction = new HashMap<>();
+                toolInteraction.put("output", interaction);
+                toolInteractions.add(toolInteraction);
+            }
+        }
+        builder.toolInteractions(toolInteractions);
+
+        Map<String, String> contextParameters = new HashMap<>();
+        contextParameters.putAll(parameters);
+        builder.parameters(contextParameters);
+
+        return builder.build();
+    }
+
+    public static Object emitPostToolHook(
+        Object toolOutput,
```

Comment:
// Validate required parameters
    if (toolOutput == null) {
        log.warn("Tool output is null, skipping POST_TOOL hook");
        return null;
    }

add validation of the toolOutput?